plans/init.pp

```diff
@@ -0,0 +1,96 @@
+# Reboots targets and waits for them to be available again.
+#
+# @param nodes Targets to reboot.
+# @param message Message to log with the reboot (for platforms that support it).
+# @param reboot_delay How long (in seconds) to wait before rebooting. Defaults to 1.
+# @param disconnect_wait How long (in seconds) to wait before checking whether the server has rebooted. Defaults to 1.
+# @param reconnect_wait How long (in seconds) to attempt to reconnect before giving up. Defaults to 120.
+# @param retry_interval How long (in seconds) to wait between retries. Defaults to 1.
+plan reboot (
+  TargetSpec $nodes,
+  Optional[String] $message = undef,
+  Integer[1] $reboot_delay = 1,
+  Integer[0] $disconnect_wait = 10,
+  Integer[0] $reconnect_wait = 120,
+  Integer[0] $retry_interval = 1,
+) {
+  $start_time = Timestamp()
+  $targets = get_targets($nodes)
+
+  # Get last boot time
+  $begin_boot_time_results = without_default_logging() || {
+    run_task('reboot::last_boot_time', $targets)
+  }
+
+  # Reboot; catch errors here because the connection may get cut out from underneath
+  $reboot_result = run_task('reboot', $nodes, timeout => $reboot_delay, message => $message)
```

Comment:
Having this be a single plan means that if most nodes successfully reboot but one fails, it's hard to recover. May need to split waiting for the reboot into a separate plan. Should we catch errors, wait for reboot on the successful nodes, then fail?

Reply:
@reidmv any input on this question?

Reply:
Catch errors -> wait  for all nodes to finish -> fail seems like a logical eventflow to me, but I don't have an exact use case in mind.